azure/functions/durable_functions.py

```diff
@@ -1,45 +1,91 @@
-from typing import Any
+import typing
+import json
+
 from azure.functions import _durable_functions
 
 from . import meta
 
 
 # Durable Function Orchestration Trigger
 class OrchestrationTriggerConverter(meta.InConverter,
+                                    meta.OutConverter,
                                     binding='orchestrationTrigger',
                                     trigger=True):
     @classmethod
     def check_input_type_annotation(cls, pytype):
         return issubclass(pytype, _durable_functions.OrchestrationContext)
 
+    @classmethod
+    def check_output_type_annotation(cls, pytype):
+        # Implicit output should accept any return type
+        return True
+
     @classmethod
     def decode(cls,
                data: meta.Datum, *,
                trigger_metadata) -> _durable_functions.OrchestrationContext:
         return _durable_functions.OrchestrationContext(data.value)
 
+    @classmethod
+    def encode(cls, obj: typing.Any, *,
+               expected_type: typing.Optional[type]) -> meta.Datum:
+        # Durable function context should be a json
+        return meta.Datum(type='json', value=obj)
+
     @classmethod
     def has_implicit_output(cls) -> bool:
         return True
 
 
 # Durable Function Activity Trigger
 class ActivityTriggerConverter(meta.InConverter,
+                               meta.OutConverter,
                                binding='activityTrigger',
                                trigger=True):
     @classmethod
     def check_input_type_annotation(cls, pytype):
         # Activity Trigger's arguments should accept any types
         return True
 
+    @classmethod
+    def check_output_type_annotation(cls, pytype):
+        # Implicit output should accept any return type
+        return True
+
     @classmethod
     def decode(cls,
                data: meta.Datum, *,
-               trigger_metadata) -> Any:
-        if getattr(data, 'value', None) is not None:
-            return data.value
+               trigger_metadata) -> typing.Any:
+        data_type = data.type
+
+        # Durable functions extension always returns a string of json
+        # See durable functions library's call_activity_task docs
+        if data_type == 'string' or data_type == 'json':
```

Comment:
Do we have a test to cover the decode for json type?

Reply:
Yes, I add some tests for in the next iteration. Thanks for the review.